File: 287-find-the-duplicate-number/287-find-the-duplicate-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findDuplicate = function(nums) {
    let found = [];
    
    for (let i = 0; i < nums.length; i++) {
        let num = nums[i];
        if ( found[num] ) return num;
        else found[num] = true;
    }
};

File: 287-find-the-duplicate-number/287-find-the-duplicate-number.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        map <int, bool> isExist;
        
        for (int i = 0; i < nums.size(); i++ ) {
            int num = nums[i];
            
            if ( isExist[num] == true ) return num;
            else isExist[num] = true;
        }
        
        return 0;
    }
};